grapple/helpers.py

```diff
@@ -104,7 +104,88 @@ def resolve_plural(self, _, info, **kwargs):
             )
 
             setattr(
-                schema, "resolve_" + field_name, MethodType(resolve_singular, schema),
+                schema, "resolve_" + field_name, MethodType(resolve_singular, schema)
+            )
+            setattr(
+                schema,
+                "resolve_" + plural_field_name,
+                MethodType(resolve_plural, schema),
+            )
+            return schema
+
+        # Send schema to Grapple schema.
+        register_graphql_schema(Mixin())
+        return cls
+
+    return inner
+
+
+def register_paginated_query_field(
+    field_name,
+    plural_field_name=None,
+    query_params=None,
+    required=False,
+    plural_required=False,
+    plural_item_required=False,
+):
+    from .types.structures import PaginatedQuerySet
+    from .utils import resolve_paginated_queryset
+
+    if not plural_field_name:
+        plural_field_name = field_name + "s"
+
+    def inner(cls):
+        field_type = lambda: registry.models[cls]
+        field_query_params = query_params or {"id": graphene.Int()}
+
+        def Mixin():
+            # Generic methods to get all and query one model instance.
+            def resolve_singular(self, _, info, **kwargs):
+                # If no filters then return nothing,
+                if not kwargs:
+                    return None
+
+                try:
+                    # If is a Page then only query live/public pages.
+                    if issubclass(cls, Page):
+                        return cls.objects.live().public().get(**kwargs)
+
+                    return cls.objects.get(**kwargs)
+                except:
+                    return None
+
+            def resolve_plural(self, _, info, **kwargs):
+                qs = cls.objects
+                if issubclass(cls, Page):
+                    qs = cls.objects.live().public().order_by("-first_published_at")
```

Comment:
`order_by` should be customisable, but let's leave that as a followup

Reply:
that's just the default, if the `order` argument is passed, this will be overwritten by the `resolve_paginated_queryset` function.

Reply:
ah, I see the `order` kwarg on `PaginatedQuerySet` 👍